src/main/java/uk/gov/justice/digital/delius/service/ReferralService.java

```diff
@@ -1,46 +1,85 @@
 package uk.gov.justice.digital.delius.service;
 
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Qualifier;
-import org.springframework.http.HttpHeaders;
-import org.springframework.http.MediaType;
-import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
-
-import org.springframework.web.reactive.function.client.WebClient;
+import uk.gov.justice.digital.delius.data.api.Nsi;
 import uk.gov.justice.digital.delius.data.api.ReferralSentRequest;
-import uk.gov.justice.digital.delius.data.api.deliusapi.NewContact;
+import uk.gov.justice.digital.delius.data.api.ReferralSentResponse;
+import uk.gov.justice.digital.delius.data.api.deliusapi.NewNsi;
+import uk.gov.justice.digital.delius.data.api.deliusapi.NewNsiManager;
+
+import java.util.Collections;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.toList;
 
 @Service
 public class ReferralService {
-    private final WebClient webClient;
+    DeliusApiClient deliusApiClient;
 
-    @Autowired
-    public ReferralService(@Qualifier("deliusApiWebClient") final WebClient webClient) {
-        this.webClient = webClient;
+    NsiService nsiService;
+
+    OffenderService offenderService;
+
+    public ReferralService(final DeliusApiClient deliusApiClient, final NsiService nsiService, final OffenderService offenderService) {
+        this.deliusApiClient = deliusApiClient;
+        this.nsiService = nsiService;
+        this.offenderService = offenderService;
     }
 
     @Transactional
-    public ResponseEntity<String> createReferralSent(final String crn,
-                                                     final ReferralSentRequest referralSent) {
+    public ReferralSentResponse createNsiReferral(final String crn,
+                                                  final ReferralSentRequest referralSent) {
+        var existingNsi = getExistingMatchingNsi(crn, referralSent);
+
+        if(existingNsi.isPresent()) {
```

Comment:
you could rewrite this as 
`return ReferralSentResponse.builder().nsiId(existingNsi.map(n -> n..getNsiId().orElseGet(() -> {
  // code to call Delius
}
)).build()`